### PR TITLE
LPS-154049

### PR DESCRIPTION
**Pull request by marcelabc**
Related Issue: https://issues.liferay.com/browse/LPS-154049

### CODE REVIEW

**Comment by liferay-forms** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
Hi, you don’t need to run the CI commands anymore, I’ll do it for you and let you know if any issue happens.

**Comment by liferay-forms** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
Hi, you don’t need to run the CI commands anymore, I’ll do it for you and let you know if any issue happens.

**Comment by liferay-forms** [comment]
Hi, you don’t need to run the CI commands anymore, I’ll do it for you and let you know if any issue happens.

**Comment by liferay-forms** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
ci:stop

**Comment by liferay-forms** [comment]
Hey @marcelabc, can you please check if these failures are related to your changes? Thanks

**Comment by liferay-forms** [comment]
Hey @marcelabc, can you please check if these failures are related to your changes? Thanks

**Comment by liferay-forms** [comment]
Hey @marcelabc, can you please check if these failures are related to your changes? Thanks

**Comment by liferay-forms** [comment]
Hey @marcelabc, can you please check if these failures are related to your changes? Thanks

**Comment by liferay-forms** [comment]
Hey @marcelabc, can you please check if these failures are related to your changes? Thanks

**Comment by liferay-forms** [comment]
Hey @marcelabc, can you please check if these failures are related to your changes? Thanks

**Comment by liferay-forms** [comment]
Hey @marcelabc, can you please check if these failures are related to your changes? Thanks

**Comment by liferay-forms** [comment]
Hey @marcelabc, can you please check if these failures are related to your changes? Thanks

**Comment by liferay-forms** [comment]
Hey @marcelabc, can you please check if these failures are related to your changes? Thanks

**Comment by liferay-forms** [comment]
Hey @marcelabc, can you please check if these failures are related to your changes? Thanks

**Comment by liferay-forms** [comment]
Hey @marcelabc, can you please check if these failures are related to your changes? Thanks

**Comment by liferay-forms** [comment]
Hey @marcelabc, can you please check if these failures are related to your changes? Thanks

**Comment by liferay-forms** [comment]
Hey @marcelabc, can you please check if these failures are related to your changes? Thanks

**Comment by liferay-forms** [comment]
Hey @marcelabc, can you please check if these failures are related to your changes? Thanks

**Comment by liferay-forms** [comment]
Hey @marcelabc, can you please check if these failures are related to your changes? Thanks

**Comment by liferay-forms** [comment]
Hey @marcelabc, can you please check if these failures are related to your changes? Thanks

**Comment by liferay-forms** [comment]
Hey @marcelabc, can you please check if these failures are related to your changes? Thanks

**Comment by liferay-forms** [comment]
Hey @marcelabc, can you please check if these failures are related to your changes? Thanks

**Comment by liferay-forms** [comment]
Hey @marcelabc, can you please check if these failures are related to your changes? Thanks

**Comment by liferay-forms** [comment]
Hey @marcelabc, can you please check if these failures are related to your changes? Thanks

**Comment by liferay-forms** [comment]
Hey @marcelabc, can you please check if these failures are related to your changes? Thanks

**Comment by liferay-forms** [comment]
Hey @marcelabc, can you please check if these failures are related to your changes? Thanks

**Comment by liferay-forms** [comment]
Hey @marcelabc, can you please check if these failures are related to your changes? Thanks

**Comment by liferay-forms** [comment]
Hey @marcelabc, can you please check if these failures are related to your changes? Thanks

**Comment by liferay-forms** [comment]
Hey @marcelabc, can you please check if these failures are related to your changes? Thanks

**Comment by liferay-forms** [comment]
Hey @marcelabc, can you please check if these failures are related to your changes? Thanks

**Comment by liferay-forms** [comment]
Hey @marcelabc, can you please check if these failures are related to your changes? Thanks

**Comment by liferay-forms** [comment]
Hey @marcelabc, can you please check if these failures are related to your changes? Thanks

**Comment by liferay-forms** [comment]
Hey @marcelabc, can you please check if these failures are related to your changes? Thanks

**Comment by liferay-forms** [comment]
Hey @marcelabc, can you please check if these failures are related to your changes? Thanks

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/instance/lifecycle/SystemObjectDefinitionMetadataPortalInstanceLifecycleListener.java [hunk 1]
```diff
@@ -219,6 +220,16 @@ private void _apply(
 						_persistedModelLocalServiceRegistry,
 						systemObjectDefinitionMetadata),
 					null);
+
```
Comment (marcelabc) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/instance/lifecycle/SystemObjectDefinitionMetadataPortalInstanceLifecycleListener.java:223]: separate FF

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/related/models/SystemObjectMtoMObjectRelatedModelsProviderImpl.java [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+
+		PersistedModelLocalService persistedModelLocalService =
+			_persistedModelLocalServiceRegistry.getPersistedModelLocalService(
+				_systemObjectDefinitionMetadata.getModelClassName());
+
+		return persistedModelLocalService.dslQueryCount(dslQuery);
+	}
+
+	private GroupByStep _getGroupByStep(
+			long groupId, long objectRelationshipId, long primaryKey,
```
Comment (marcelabc) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/related/models/SystemObjectMtoMObjectRelatedModelsProviderImpl.java:295]: pass the objectRelationship entity

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/related/models/SystemObjectMtoMObjectRelatedModelsProviderImpl.java [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+			long groupId, long objectRelationshipId, long primaryKey)
+		throws PortalException {
+
+		ObjectRelationship objectRelationship =
+			_objectRelationshipLocalService.getObjectRelationship(
+				objectRelationshipId);
+
+		boolean reverse = objectRelationship.isReverse();
+
+		if (objectRelationship.isReverse()) {
```
Comment (marcelabc) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/related/models/SystemObjectMtoMObjectRelatedModelsProviderImpl.java:179]: remove this condition

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/related/models/SystemObjectMtoMObjectRelatedModelsProviderImpl.java [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+			int end)
+		throws PortalException {
+
+		ObjectRelationship objectRelationship =
+			_objectRelationshipLocalService.getObjectRelationship(
+				objectRelationshipId);
+
+		boolean reverse = objectRelationship.isReverse();
+
+		if (objectRelationship.isReverse()) {
```
Comment (marcelabc) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/related/models/SystemObjectMtoMObjectRelatedModelsProviderImpl.java:148]: remove this condition

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/related/models/SystemObjectMtoMObjectRelatedModelsProviderImpl.java [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+			objectRelationship =
+				_objectRelationshipLocalService.fetchReverseObjectRelationship(
+					objectRelationship, false);
+		}
+
+		DSLQuery dslQuery = _getGroupByStep(
```
Comment (marcelabc) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/related/models/SystemObjectMtoMObjectRelatedModelsProviderImpl.java:154]: INLINE

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/related/models/SystemObjectMtoMObjectRelatedModelsProviderImpl.java [hunk 6]
```diff
@@ -0,0 +1,289 @@
...
```
Comment (marcelabc) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/related/models/SystemObjectMtoMObjectRelatedModelsProviderImpl.java:185]: inline

**Comment by marcelabc** [comment]
ci:stop

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/related/models/SystemObjectMtoMObjectRelatedModelsProviderImpl.java [hunk 7]
```diff
@@ -196,6 +196,101 @@ public int getRelatedModelsCount(
 		return persistedModelLocalService.dslQueryCount(dslQuery);
 	}
 
+	@Override
+	public List<T> getUnrelatedModels(
+			long companyId, long groupId, ObjectDefinition objectDefinition,
+			long objectEntryId, long objectRelationshipId)
+		throws PortalException {
+
+		Column<?, Long> companyIdColumn = (Column<?, Long>)_table.getColumn(
```
Comment (marcelabc) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/related/models/SystemObjectMtoMObjectRelatedModelsProviderImpl.java:205]: see groupByStep

**Pull request closed without merging**